streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -397,7 +397,7 @@
      * @param suppressed Configuration object determining what, if any, updates to suppress
      * @return A new KTable with the desired suppression characteristics.
      */
-    KTable<K, V> suppress(final Suppressed<K> suppressed);
+    KTable<K, V> suppress(final Suppressed<? super K> suppressed);
```

Comment:
Without this change, final-results configs that also set a name become awkward, as you have to redundantly specify the type parameters at the call site (like `Suppressed.<Windowed<...>>untilWindowCloses(...).withName(...)`)

This change, in conjunction with the change to untilWindowCloses in Suppressed to drop the generic parameter makes this a non-issue (as you can tell in the tests).